staging/src/k8s.io/client-go/tools/watch/until_test.go

```diff
@@ -209,7 +209,7 @@ func TestUntilWithSync(t *testing.T) {
 			conditionFunc: func(e watch.Event) (bool, error) {
 				return true, nil
 			},
-			expectedErr:   wait.ErrWaitTimeout,
+			expectedErr:   wait.ErrorInterrupted(errors.New("timed out waiting for the condition")),
```

Comment:
I think you can return `wait.ErrorInterrupted()` directly because it's compatible. see 

```
func (e errInterrupted) Error() string {
	if e.cause == nil {
		// returns the same error message as historical behavior
		return "timed out waiting for the condition"
	}
	return e.cause.Error()
}
```

Reply:
okay,done